libraries/c_sdk/standard/mqtt/test/system/iot_tests_mqtt_system.c

```diff
@@ -211,6 +211,46 @@ static bool _disconnectSerializerOverride = false;  /**< @brief Tracks if #_disc
 
 /*-----------------------------------------------------------*/
 
+/**
+ * @brief Establish an MQTT connection. Retry if enabled.
+ */
+static IotMqttError_t _mqttConnect( const IotMqttNetworkInfo_t * pNetworkInfo,
```

Comment:
At the top of this file (before line 62) could you add:

```c
#ifndef IOT_TEST_MQTT_CONNECT_RETRY_COUNT
#define IOT_TEST_MQTT_CONNECT_RETRY_COUNT 1
#endif
```

And also before line 70:
```c
#if IOT_TEST_MQTT_CONNECT_RETRY_COUNT < 1
#error "IOT_TEST_MQTT_CONNECT_RETRY_COUNT must be at least 1."
#endif
```

You can then remove all the preprocessor directives from this function and just have a for loop.